pkg/server/wire.go

```diff
@@ -334,8 +334,7 @@ var wireBasicSet = wire.NewSet(
 	secretsMigrations.ProvideDataSourceMigrationService,
 	secretsMigrations.ProvideMigrateToPluginService,
 	secretsMigrations.ProvideMigrateFromPluginService,
-	secretsMigrations.ProvideSecretMigrationService,
-	wire.Bind(new(secretsMigrations.SecretMigrationService), new(*secretsMigrations.SecretMigrationServiceImpl)),
```

Comment:
I might be confused about wire, but don't we still need the bind here?

From the docs:

```Wire matches inputs to outputs via [type identity](https://golang.org/ref/spec#Type_identity), so the inclination might be to create a provider that returns an interface type. However, this would not be idiomatic, since the Go best practice is to [return concrete types](https://github.com/golang/go/wiki/CodeReviewComments#interfaces). Instead, you can declare an interface binding in a provider set:```

Reply:
Oh! Will sitck to concrete types from now on, changed the return type to interface, will revert it back and add the wire.Bind.

Reply:
Done on https://github.com/grafana/grafana/pull/54676/commits/4af903d7a4676ca432a342d703cb5d19b6f4153f